home_energy_management/device_simulators/electric_vehicle.py

```diff
@@ -133,83 +132,105 @@ def update(self, info: InfoForDevice) -> DeviceResponse:
-
     def get_info(self) -> dict[str, Any]:
         now = self.get_time()
-        return {
-            "driving_power": self.get_driving_power(now),
-            "time_until_charged": self.get_time_until_charged(now),
-        }
+        return {"driving_power": self.get_driving_power(now)}
 
     def set_params(self, params: dict[str, Any]) -> None:
         pass
 
 
-class LiveEVDriving(AbstractEVDriving):
+class LiveEVDriving(EVDriving):
     driving_power: float
-    next_ev_charged_time: int
 
-    def __init__(
-            self,
-            driving_power: float,
-            time_until_charged_h: Optional[float] = None):
-        AbstractEVDriving.__init__(self)
-        self.driving_power = driving_power
-        self.next_ev_charged_time = int(time_until_charged_h * 3600)
+    def __init__(self, driving_power: float,):
+        EVDriving.__init__(self)
+        self.driving_power = float(driving_power)
+
+    def set_driving_power(self, driving_power: float):
+        self.driving_power = float(driving_power)
 
     def get_driving_power(self, now: int):
         return self.driving_power
 
-    def get_time_until_charged(self, now: int) -> int:
-        left_time = self.next_ev_charged_time - now
-        return -1 if (left_time <= 0
-                      or self.driving_power > 0
-                      or self.next_ev_charged_time == METERSIM_NO_UPDATE_SCHEDULED) else left_time
-
     def get_info(self) -> dict[str, Any]:
-        now = self.get_time()
-        return {
-            "driving_power": self.driving_power,
-            "time_until_charged": self.get_time_until_charged(now),
-        }
+        return {"driving_power": self.driving_power}
 
     def set_params(self, params: dict[str, Any]) -> None:
         pass
 
-    def update_state(self,
-                     driving_power: float,
-                     time_until_charged_h: Optional[float] = None):
-        self.driving_power = driving_power
-        if time_until_charged_h:
-            self.next_ev_charged_time = self.get_time() + int(time_until_charged_h * 3600)
-        else:
-            self.next_ev_charged_time = METERSIM_NO_UPDATE_SCHEDULED
+
+class EVDeparturePlans(Device, ABC):
+    @abstractmethod
+    def get_time_until_departure(self) -> float:
+        pass
+
+    def update(self, info: InfoForDevice) -> DeviceResponse:
+        return DeviceResponse([0.0, 0.0, 0.0], METERSIM_NO_UPDATE_SCHEDULED)
+
+
+class ScheduledEVDeparturePlans(EVDeparturePlans, ScheduledDevice, Device):
+    def __init__(
+            self,
+            config: list[tuple[int, Any]],
+            loop: int = 0):
+        EVDeparturePlans.__init__(self)
+        ScheduledDevice.__init__(self, config, loop)
+
+    def get_time_until_departure(self) -> int:
+        now = self.get_time()
+        _, next_update_time = self.get_state(now)
+        update_driving_power = 0.
+        time = now
```

Comment:
Ta linijka nie jest potrzebna

Reply:
Masz na myśli linijkę 205 (bo komentarz jej dodany do 202-205)? Faktycznie nie jest potrzebna - dodałam ją tylko dlatego, że inaczej pycharm ostrzega, że: "Local variable 'time' might be referenced before assignment", choć przy ustawieniu "update_driving_power = 0." jest to niemożliwe, więc usuwam